### PR TITLE
100% code coverage

### LINKED ISSUE #56: Fix Missing Code Coverage
Ensure we're at 100% code coverage for our specs!

Also, mandate 100% branch coverage (see #59 for more context)

### PR DESCRIPTION
**Pull request by jemmaissroff**
Adds specs to ensure we're at 100% code coverage, and mandates 100% branch coverage moving forward.

Resolves #56

### CODE REVIEW

File: spec/memo_wise_spec.rb [hunk 1]
```diff
@@ -276,19 +274,45 @@ def unmemoized_method
       expect(instance2.no_args_counter).to eq(0)
     end
 
-    it "keeps private methods private" do
-      expect(instance.private_methods.include?(:private_memowise_method)).
-        to eq(true)
+    context "with private methods" do
+      it "keeps private methods private" do
+        expect(instance.private_methods.include?(:private_memowise_method)).
```
Comment (JacobEvelyn) [inline spec/memo_wise_spec.rb:279]: This might be clearer as

```ruby
expect(instance.private_methods).to include(: private_memowise_method)
```

(And same comment below.) Thoughts?

File: spec/memo_wise_spec.rb [hunk 2]
```diff
@@ -396,6 +420,24 @@ def no_args
...
+
+        expect(Array.new(4) do
+          instance.with_positional_and_keyword_args(1, b: 2)
+        end).to all eq("with_positional_and_keyword_args: a=1, b=2")
+
+        expect(Array.new(4) do
+          instance.with_positional_and_keyword_args(2, b: 3)
+        end).to all eq("with_positional_and_keyword_args: a=2, b=3")
+
+        # This should be executed twice for each set of arguments passed,
```
Comment (JacobEvelyn) [inline spec/memo_wise_spec.rb:436]: This comment is a little confusing—it's really once for each set of arguments passed, which is twice total. Could you rephrase this?

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
   # SimpleCov.refuse_coverage_drop is only implemented for line coverage, so for
   # branch coverage we must use `minimum_coverage`
-  SimpleCov.minimum_coverage branch: 90
+  SimpleCov.minimum_coverage branch: 100
```
Comment (JacobEvelyn) [inline spec/spec_helper.rb:23]: 🎉

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -20,7 +20,7 @@
...
 
   SimpleCov.refuse_coverage_drop
```
Comment (JacobEvelyn) [inline spec/spec_helper.rb:25]: I don't have strong feelings, but it might be more explicit if we changed this to also be `100`. Happy either way though!

Comment (jemmaissroff) [reply spec/spec_helper.rb:25]: Going to leave as is for now!

I have [a PR up on SimpleCov](https://github.com/simplecov-ruby/simplecov/pull/934) to hopefully get branch coverage working for `:refuse_coverage_drop`, so hoping to just use refuse_coverage_drop for both eventually.

**Review by ms-ati — APPROVED** [review]
LGTM
